data/publiccloud/terraform/azure.tf

```diff
@@ -157,59 +167,57 @@ resource "azurerm_image" "image" {
-    vm_size               = var.type
-    count                 = var.instance_count
-
-    storage_image_reference {
-        id = var.image_id != "" ? azurerm_image.image.0.id : ""
-        publisher = var.image_id != "" ? "" : "SUSE"
-        offer     = var.image_id != "" ? "" : var.offer
-        sku       = var.image_id != "" ? "" : var.sku
-        version   = var.image_id != "" ? "" : "latest"
-    }
+resource "azurerm_linux_virtual_machine" "openqa-vm" {
+  name                = "${var.name}-${element(random_id.service.*.hex, count.index)}"
+  resource_group_name = azurerm_resource_group.openqa-group.name
+  location            = var.region
+  size                = var.type
+  computer_name  = "${var.name}-${element(random_id.service.*.hex, count.index)}"
+  admin_username      = "azureuser"
+  disable_password_authentication = true
 
-    storage_os_disk {
-        name              = "${var.name}-${element(random_id.service.*.hex, count.index)}-osdisk"
-        caching           = "ReadWrite"
-        create_option     = "FromImage"
-        managed_disk_type = "Standard_LRS"
-    }
+  count                 = var.instance_count
 
-    os_profile {
-        computer_name  = "${var.name}-${element(random_id.service.*.hex, count.index)}"
-        admin_username = "azureuser"
-    }
+  network_interface_ids = [azurerm_network_interface.openqa-nic[count.index].id]
 
-    os_profile_linux_config {
-        disable_password_authentication = true
-        ssh_keys {
-            path     = "/home/azureuser/.ssh/authorized_keys"
-            key_data = file("/root/.ssh/id_rsa.pub")
-        }
-    }
+  tags = merge({
+          openqa_created_by = var.name
+          openqa_created_date = timestamp()
+          openqa_created_id = element(random_id.service.*.hex, count.index)
+      }, var.tags)
 
-    tags = merge({
-            openqa_created_by = var.name
-            openqa_created_date = timestamp()
-            openqa_created_id = element(random_id.service.*.hex, count.index)
-        }, var.tags)
+  admin_ssh_key {
+    username   = "azureuser"
+    public_key = file("/root/.ssh/id_rsa.pub")
+  }
+
+  os_disk {
+    name              = "${var.name}-${element(random_id.service.*.hex, count.index)}-osdisk"
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
 
+  source_image_id =  var.image_id != "" ? azurerm_image.image.0.id : ""
+  source_image_reference {
+    publisher = lookup(var.image, "publisher")
+    offer     = lookup(var.image, "offer")
+    sku       = lookup(var.image, "sku")
+    version   = lookup(var.image, "version")
+#    publisher = var.image_id != "" ? "" : "SUSE"
+#    offer     = var.image_id != "" ? "" : var.offer
+#    sku       = var.image_id != "" ? "" : var.sku
+#    version   = var.image_id != "" ? "" : "latest"
+  }
 
-    boot_diagnostics {
-        enabled = true
-        storage_uri = "https://${var.storage-account}.blob.core.windows.net/"
-    }
+  boot_diagnostics {
```

Comment:
Do we really need the boot_diagnostics for our test runs?

The biggest advantage of Azure boot diagnostics is to have the serial output. Since this is not used in our automated test runs at all, I don't see why we need this.

Reply:
I agree to solve it in separate ticket. You may create one.